Mage.Sets/src/mage/cards/l/LegateLaniusCaesarsAce.java

```diff
@@ -0,0 +1,102 @@
+package mage.cards.l;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SacrificePermanentTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeOpponentsEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetSacrifice;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author notgreat
+ */
+public final class LegateLaniusCaesarsAce extends CardImpl {
+
+    public LegateLaniusCaesarsAce(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}{R}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Decimate -- When Legate Lanius enters the battlefield, each opponent sacrifices a tenth of the creatures they control, rounded up.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new LegateLaniusCaesarsAceSacrificeEffect()).withFlavorWord("Decimate"));
+
+        // Whenever an opponent sacrifices a creature, put a +1/+1 counter on Legate Lanius.
+        this.addAbility(new SacrificePermanentTriggeredAbility(Zone.BATTLEFIELD,
+                new AddCountersSourceEffect(CounterType.P1P1.createInstance()),
+                StaticFilters.FILTER_PERMANENT_A_CREATURE, TargetController.OPPONENT, SetTargetPointer.NONE, false
+        ));
+    }
+
+    private LegateLaniusCaesarsAce(final LegateLaniusCaesarsAce card) {
+        super(card);
+    }
+
+    @Override
+    public LegateLaniusCaesarsAce copy() {
+        return new LegateLaniusCaesarsAce(this);
+    }
+}
+
+//Based on SacrificeAllEffect
+class LegateLaniusCaesarsAceSacrificeEffect extends OneShotEffect {
```

Comment:
Try to use standard effect `SacrificeOpponentsEffect` with `HalfValue` dynamic values instead custom. Also you can add card hint with that value like "You would have to sacrifice: xxx"

Reply:
The current `SacrificeOpponentsEffect` calculates the number once at the start. I could move it to inside the loop, but the dynamic value's calculate function still has no good way of knowing which opponent is being evaluated. I suppose I could smuggle the information in as a local variable, but that felt like a sufficiently messy hack that it shouldn't be added to the common class for just one card.

Reply:
Agree that custom class is fine here